Dev

DenisFilichev opened the pull request.

Commit:
Первый набросок программы

Commit:
Добавлено сохранение товароы и заказов в файлы

Commit:
1. Разделены классы, работающие с файлами .dat, по типу Product и Order
2. Сделана корректировка количества товара на складе при его добавлении
в заказ и удалении из заказа.
3. Сделана часть проверок на ошибки.

Commit:
1. Разделены классы, работающие с файлами .dat, по типу Product и Order
2. Сделана корректировка количества товара на складе при его добавлении
в заказ и удалении из заказа.
3. Сделана часть проверок на ошибки.

Commit:
1. Разделены классы, работающие с файлами .dat, по типу Product и Order
2. Сделана корректировка количества товара на складе при его добавлении
в заказ и удалении из заказа.
3. Сделана часть проверок на ошибки.
4. Грфическое окно

Commit:
Сделал отображение товаров заказа в нижнем поле SplitPane

Commit:
Сделано:
1. Добавление нового заказа
2. Добавление нового товара на склад
3. Добавление товара в заказ
4. Удаление заказа с корректировкой склада
5. Удаление товара из заказа с корректировкой склада

Commit:
1. Реализована проверка на дублирование добавляемого в заказ товара.
2. Сделана проверка соответствия вводимых полей нового товара.

Commit:
Включил запись / чтение файлов

konivex commented on src/ru/avalon/java/dev/j12/labs/models/Order.java:
```diff
@@ -0,0 +1,126 @@
+        int index = SearchByID.objectSearch(ID, list);
+        Product prod = list.get(index);
+        if (prod.getBalance() >= quantity && quantity > 0){
+            Product prodOrder = new Product(prod.getID(), prod.getName(), prod.getColor());
```
Зачем создавать копию продукта, если можно запомнить ссылку на оригинал?

DenisFilichev replied:
Я целенаправлено делаю копию товара, так сказать слепок приобретаемого товара и последующие возможные изменения корневого товара ни как не отразятся на истории сделанных ранее заказов. Т.е., например, со временем поменяется цена товара или его описание и это ни как не отразится на уже проданных экземплярах данного товара.

konivex commented on src/ru/avalon/java/dev/j12/labs/controlers/FileListOrders.java:
```diff
@@ -0,0 +1,36 @@
+ *
+ * @author denis
+ */
+public class FileListOrders {
```
Исправьте этот класс и FileListProducts так, чтобы их методы были instance, а не static.

DenisFilichev replied:
Сделано

konivex commented on src/ru/avalon/java/dev/j12/labs/Application.java:
```diff
@@ -14,42 +19,65 @@
+            productListObject = FileListProducts.fileRead();
+        }catch (IOException | ClassNotFoundException e){
+        }catch (NullPointerException r){System.out.println("NullPointerException");}
+        
```
Если что-то не получилось прочитать, то надо вывести human-readable сообщение об этом и завершить работу..

DenisFilichev replied:
Сделано

konivex commented on src/ru/avalon/java/dev/j12/labs/Application.java:
```diff
@@ -14,42 +19,65 @@
         
+        //Записываем файл со списком товаров
+        try {FileListOrders.fileWrite(orderListObject);
+        } catch (IOException e) {}
```
Пользователя нужно проинформировать и наличии проблем с сохранением результатов его работы.

DenisFilichev replied:
Сделано

Commit:
Включил запись / чтение файлов - устранен баг с сохранением изменений

konivex commented on src/ru/avalon/java/dev/j12/labs/controlers/SearchByID.java:
```diff
@@ -0,0 +1,32 @@
+    public static <T extends IDSearch>int objectSearch (int ID, ArrayList <T> list){
+        int i = -1;
+        if (list.isEmpty()) return i;
+        for (Object obj : list){
```
Вместо `Object obj` можно написать `T obj`, тогда вместо `if (((T)obj).getID()==ID)` можно будет писать `if (obj.getID()==ID)`.

DenisFilichev replied:
Сделано

konivex reviewed (changes requested):
Графический интерфейс тоже можно смотреть?

Commit:
1. Методы в классах FileListProducts и FileListOrders исправлены на instance
2. Сделано информирование пользователя об ошибках возникающих при сохранении или чтении файлов .dat
3. Сделаны мелкие исправления в классе SearchByID

konivex reviewed (approved):
Зачтено.
